src/structure_managers/adaptor_neighbour_list.hh

```diff
@@ -807,33 +845,36 @@ namespace rascal {
    */
   template <class ManagerImplementation>
   void AdaptorNeighbourList<ManagerImplementation>::update_self() {
-    // set the number of centers
-    this->n_centers = this->manager->get_size();
-    this->n_ghosts = 0;
-    //! Reset cluster_indices for adaptor to fill with sequence
-    internal::for_each(this->cluster_indices_container,
-                       internal::ResizePropertyToZero());
-
-    // initialize necessary data structure
-    this->atom_tag_list.clear();
-    this->atom_types.clear();
-    this->ghost_atom_tag_list.clear();
-    this->nb_neigh.clear();
-    this->neighbours_atom_tag.clear();
-    this->offsets.clear();
-    this->ghost_positions.clear();
-    this->ghost_types.clear();
-    // actual call for building the neighbour list
-    this->make_full_neighbour_list();
-    this->set_offsets();
-
-    // layering is started from the scratch, therefore all clusters and
-    // centers+ghost atoms are in the right order.
-    auto & atom_cluster_indices{std::get<0>(this->cluster_indices_container)};
-    auto & pair_cluster_indices{std::get<1>(this->cluster_indices_container)};
-
-    atom_cluster_indices.fill_sequence(this->consider_ghost_neighbours);
-    pair_cluster_indices.fill_sequence();
+    if (this->need_update) {
```

Comment:
Did someone check how much this actually helps?
On the lammps side, this will be unnecessary. Is it related to PLUMED, why this was added?
I mean we are not building a an integrator here, are we?

Reply:
lammps does not use adaptor NL so it is unrelated. We just don't want to update the linked cell if it is not needed folloing the skin criteria